### PR TITLE
 chore: simplify `index.ts`

### PR DESCRIPTION
**Pull request by DominicKramer**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -133,67 +126,85 @@ export class ErrorReporting {
...
+  /**
+   * @example
+   * const hapi = require('hapi');
+   * const server = new hapi.Server();
+   * server.connection({ port: 3000 });
+   * server.start();
+   * // AFTER ALL OTHER ROUTE HANDLERS
+   * server.register({register: errors.hapi});
+   */
+  hapi() {
```
Comment (kjin) [inline src/index.ts]: Isn't this changing the interface? It seems like `hapi` used to be of type `T` but is now `() => T` (where `T` is the return type of `makeHapiPlugin`). Am I missing something here?

Comment (DominicKramer) [reply src/index.ts]: No the return type of `makeHapiPlugin` is still `T`.

Comment (DominicKramer) [reply src/index.ts]: Yes, I need to address this.  I am also addressing some other issues in the system tests that are related to this.

Comment (DominicKramer) [reply src/index.ts]: Done

File: src/index.ts [hunk 2]
```diff
@@ -166,34 +135,72 @@ export class ErrorReporting {
      * // AFTER ALL OTHER ROUTE HANDLERS
      * server.register({register: errors.hapi});
      */
-    this.hapi = hapi(this._client, this._config);
+    this.hapi = hapiInterface.makeHapiPlugin(this._client, this._config);
```
Comment (kjin) [inline src/index.ts]: Why is it that `hapi` is a member field, but other plugins are not?

Comment (DominicKramer) [reply src/index.ts]: The `hapi` property is an Object while the others are functions.

File: src/index.ts (hunk 1 above)
Comment (kjin) [reply src/index.ts]: It seems like you changed `hapi` but my comment applies to all of the web frameworks.

File: src/index.ts [hunk 3]
```diff
@@ -166,34 +135,72 @@ export class ErrorReporting {
...
+  }
+
+  /**
+   * @example
+   * const koa = require('koa');
+   * const app = koa();
+   * // BEFORE ALL OTHER ROUTE HANDLERS HANDLERS
+   * app.use(errors.koa);
+   */
+  * koa(next: {}) {
```
Comment (kjin) [inline src/index.ts]: It looks like `next` is being ignored?

File: src/index.ts [hunk 4]
```diff
@@ -166,34 +135,72 @@ export class ErrorReporting {
...
+  /**
+   * @example
+   * const restify = require('restify');
+   * const server = restify.createServer();
+   * // BEFORE ALL OTHER ROUTE HANDLERS
+   * server.use(errors.restify(server));
+   */
+  restify(server: {}) {
+    return restifyInterface.handlerSetup(this._client, this._config)(
+        this._client, this._config, server);
```
Comment (kjin) [inline src/index.ts]: Are these arguments supposed to be passed in twice?

**Comment by DominicKramer** [comment]
This is ready for another look.

File: package.json [hunk 5]
```diff
@@ -54,21 +54,21 @@
   },
   "dependencies": {
     "@google-cloud/common": "^0.16.0",
+    "is": "^3.2.0",
+    "lodash.has": "^4.5.2"
+  },
+  "devDependencies": {
```
Comment (kjin) [inline package.json]: Nit: Sort `devDependencies` in alphabetical order.

**Pull request merged**